delivery_auto_refresh/models/sale_order.py

```diff
@@ -12,14 +12,31 @@ class SaleOrder(models.Model):
     carrier_id = fields.Many2one(
         compute="_compute_carrier_id", store=True, readonly=False
     )
+    available_carrier_ids = fields.Many2many(
+        comodel_name="delivery.carrier", compute="_compute_available_carrier_ids",
+    )
 
     @api.depends("partner_id", "partner_shipping_id")
     def _compute_carrier_id(self):
         if hasattr(super(), "_compute_carrier_id"):
             super()._compute_carrier_id()
         for order in self:
             action = order.action_open_delivery_wizard()
-            order.carrier_id = action["context"]["default_carrier_id"]
+            carrier_id = self.env["delivery.carrier"].browse(
+                action["context"]["default_carrier_id"]
+            )
+            # If the carrier isn't allowed for the current shipping address, we wont
+            # default to it.
+            order.carrier_id = carrier_id.filtered(
```

Comment:
But if non allowed, it will empty the field. Isn't better to leave the previous one?

Reply:
Ok, but I'm going to filter both so if any of them is allowed none will be set. Take a look on this new implementation